src/ngx_http_set_hashed_upstream.c

```diff
@@ -129,10 +178,35 @@ ngx_http_set_hashed_upstream(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
         return NGX_CONF_ERROR;
     }
 
+    hashConf = ngx_pcalloc(cf->pool,sizeof(ngx_http_set_hashed_upstream_conf));
+    hashConf->ul = ul;
+    hashConf->hashType = hashType;
+    if(ul->nelts <= 0) {
+        hashConf->nodeLen = 0;
+        hashConf->hashNodes = NULL;
+    } else {
+        hashConf->nodeLen = ul->nelts;
+        hashConf->hashNodes = ngx_pcalloc(cf->pool, sizeof(ngx_http_set_hashed_upstream_consistent_hash_node) *
+                                                    hashConf->nodeLen);
+        for (i = 0; i < hashConf->nodeLen; i++) {
+            hashConf->hashNodes[i].name = ul->elts[i];
+            hashConf->hashNodes[i].hash = ngx_hash_key_lc(hashConf->hashNodes[i].name->data,
+                                                          hashConf->hashNodes[i].name->len);
+        }
+
+        qsort(hashConf->hashNodes, hashConf->nodeLen, sizeof(ngx_http_set_hashed_upstream_consistent_hash_node),
+              (const void *) ngx_http_set_misc_consistent_hash_compare);
```

Comment:
When we have a large of nodes, (real node num * 160), this can be bad performance cause this may be called per request.